app/src/main/java/org/oppia/app/topic/practice/TopicPracticeFragmentPresenter.kt

```diff
@@ -4,98 +4,143 @@ import android.view.LayoutInflater
+          else -> throw IllegalArgumentException("Encountered unexpected view model: $viewModel")
+        }
+      }
+      .registerViewDataBinder(
+        viewType = ViewType.VIEW_TYPE_HEADER,
+        inflateDataBinding = TopicPracticeHeaderViewBinding::inflate,
+        setViewModel = TopicPracticeHeaderViewBinding::setViewModel,
+        transformViewModel = { it as TopicPracticeHeaderViewModel }
+      )
+      .registerViewDataBinder(
+        viewType = ViewType.VIEW_TYPE_SKILL,
+        inflateDataBinding = TopicPracticeSkillViewBinding::inflate,
+        setViewModel = this::bindSkillView,
+        transformViewModel = { it as TopicPracticeSkillSummaryViewModel }
+      )
+      .registerViewDataBinder(
+        viewType = ViewType.VIEW_TYPE_FOOTER,
+        inflateDataBinding = TopicPracticeFooterViewBinding::inflate,
+        setViewModel = this::bindFooterView,
+        transformViewModel = { it as TopicPracticeFooterViewModel }
+      )
+      .build()
   }
 
-  private fun subscribeToTopicLiveData() {
-    topicLiveData.observe(fragment, Observer<Topic> { result ->
-      skillSelectionAdapter.setSkillList(result.skillList)
-      skillSelectionAdapter.setSelectedSkillList(selectedSkillIdList)
-    })
-  }
-
-  private fun getTopicList(): LiveData<Topic> {
-    return Transformations.map(topicResultLiveData, ::processTopicResult)
+  private fun bindSkillView(
+    binding: TopicPracticeSkillViewBinding,
+    model: TopicPracticeSkillSummaryViewModel
+  ) {
+    binding.viewModel = model
+    binding.setVariable(BR.isChecked, selectedSkillIdList.contains(model.skillSummary.skillId))
+    binding.root.skill_check_box.setOnCheckedChangeListener { _, isChecked ->
+      if (isChecked) {
+        skillSelected(model.skillSummary.skillId)
+      } else {
+        skillUnselected(model.skillSummary.skillId)
+      }
+    }
   }
 
-  private fun processTopicResult(topic: AsyncResult<Topic>): Topic {
-    if (topic.isFailure()) {
-      logger.e("TopicPracticeFragment", "Failed to retrieve topic", topic.getErrorOrNull()!!)
-    }
-    return topic.getOrDefault(Topic.getDefaultInstance())
+  private fun bindFooterView(
+    binding: TopicPracticeFooterViewBinding,
+    model: TopicPracticeFooterViewModel
+  ) {
+    binding.viewModel = model
+    binding.setVariable(BR.isSubmitButtonActive, selectedSkillIdList.isNotEmpty())
   }
 
   private fun getTopicPracticeViewModel(): TopicPracticeViewModel {
     return viewModelProvider.getForFragment(fragment, TopicPracticeViewModel::class.java)
   }
 
+  private enum class ViewType {
+    VIEW_TYPE_HEADER,
+    VIEW_TYPE_SKILL,
+    VIEW_TYPE_FOOTER
+  }
+
   override fun skillSelected(skillId: String) {
     if (!selectedSkillIdList.contains(skillId)) {
       selectedSkillIdList.add(skillId)
     }
-    getTopicPracticeViewModel().notifySelectedSkillList(selectedSkillIdList)
+    binding.topicPracticeSkillList.apply {
+      this.adapter?.notifyDataSetChanged()
```

Comment:
The code `      this.adapter?.notifyDataSetChanged()
` is working fine when there are few items in the skill list but sometimes crashes or not performing correctly when the items increased such that user has to scroll through the list.

This code gives error "Cannot call this method while RecyclerView is computing a layout or scrolling"  sometimes when user click on the checkbox and startScrolling. I have tried applying Runnable as a solution suggested in stackoverflow but then the Start button remain unclickable no matter how many items are selected.

Reply:
removed `notifyDataSetChanged ` call and directly used TopicPracticeFooterViewBinding to update the `isSubmitButtonActive` boolean. Issue resolved.

Reply:
I have mentioned some changes to presenter file. Maybe that can help.

Reply:
Ok I will look into it.